src/d3a/models/strategy/pv.py

```diff
@@ -186,3 +194,21 @@ def event_offer(self, *, market_id, offer):
         # if offer was deleted but not traded, free the energy in state.available_energy_kWh again
         if offer.seller == self.owner.name:
             self.state.available_energy_kWh[market.time_slot] -= offer.energy
+
+    def _set_alternative_pricing_scheme(self, market):
```

Comment:
Cant this be done only once at the __init__ or when activating? Why does it have to be done for each market? Th same applies to the storage and load.